core/src/main/java/org/apache/calcite/rex/RexLiteral.java

```diff
@@ -970,7 +970,7 @@ public Comparable getValue4() {
       }
       break;
     case DECIMAL:
-      if (clazz == Long.class) {
+      if (clazz == Long.class || clazz == long.class) {
```

Comment:
@Aaaaaaron - How does such logic work?

I tested against following case on Java 8:
```
long.class.cast(1L);
```
Then I encountered ClassCastException:
```
Exception in thread "main" java.lang.ClassCastException: Cannot cast java.lang.Long to long
```

Are we on different platforms?

Reply:
The best way to confirm this works is to add unit tests...